# HG changeset patch
# User Rik <rik@octave.org>
# Date 1456428329 28800
#      Thu Feb 25 11:25:29 2016 -0800
# Node ID ea31a050bdd8632a488f27f1a71be14f4351d93c
# Parent  e5a63df8cf3a59740d11de6a24804de4af2fd5e0
Execute commands in startup.m at start for compatibility with Matlab.

* NEWS: Announce new behavior.

* basics.txi: Document file

* version-rcfile: Add command to execute startup.m if it exists.
Clean up formatting of file.

* __finish__.m: Fix typo in docstring.  Clean up comments to match
version-rcfile.

* site-rcfile:  Clean up formatting.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -17,16 +17,21 @@ Summary of important user-visible change
     Examples: 0b101 == 5 or 0B1100_0001 == 0xC1
 
  ** The default colormap is now set to "viridis" which is also the
     default colormap in matplotlib.  This new colormap fixes some of the
     main issues with the old default colormap "jet" such as its bad
     "luminance profile" and is also more similar to Matlab's new default
     colormap "parula".
 
+ ** When starting, Octave now looks in the function path for a file
+    startup.m and executes any commands found there.  This change was made
+    to accommodate Matlab users.  Octave has it's own configuration
+    system based on the file .octaverc which is preferred.
+
  ** Octal ('\NNN') and hex ('\xNN') escape sequences in single quoted
     strings are now interpreted by the function do_string_escapes().
     The *printf family of functions now supports octal and hex escape
     sequences in single-quoted strings for Matlab compatibility.
 
  ** Special octal and hex escape sequences for the pattern and
     replacement strings in regular expressions are now interpreted for
     Matlab compatibility.
diff --git a/doc/interpreter/basics.txi b/doc/interpreter/basics.txi
--- a/doc/interpreter/basics.txi
+++ b/doc/interpreter/basics.txi
@@ -369,16 +369,21 @@ Octave environment.
 This file can be used to make changes to the default Octave environment
 for a particular project.  Octave searches for this file in the current
 directory after it reads @file{~/.octaverc}.  Any use of the @code{cd}
 command in the @file{~/.octaverc} file will affect the directory where
 Octave searches for @file{.octaverc}.
 
 If you start Octave in your home directory, commands from the file
 @file{~/.octaverc} will only be executed once.
+
+@item startup.m
+This file is used to make personal changes to the default
+Octave environment.  It is executed for @sc{matlab} compatibility, but
+@file{~/.octaverc} is the preferred location for configuration changes.
 @end table
 
 A message will be displayed as each of the startup files is read if you
 invoke Octave with the @option{--verbose} option but without the
 @option{--silent} option.
 
 @node Quitting Octave
 @section Quitting Octave
diff --git a/scripts/startup/__finish__.m b/scripts/startup/__finish__.m
--- a/scripts/startup/__finish__.m
+++ b/scripts/startup/__finish__.m
@@ -16,26 +16,25 @@
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn {} {} __finish__
 ## Check for the existence of the function/script, @file{finish}, in the
 ## path or current working directory and execute it.
 ##
-## This function is intended to be excecuted upon a clean exit from Octave.
+## This function is intended to be executed upon a clean exit from Octave.
 ## This is accomplished in the system script @file{startup/octaverc} by use of
 ## the built-in function @code{atexit}.
 ## @seealso{atexit}
 ## @end deftypefn
 
 ## No function declaration, this is is an Octave script.  This means we are
 ## still in the base workspace with access to all user variables.
 
 if (exist ("finish", "file"))
-  ## No arg list here since finish might be a script.
-  finish;
+  finish;  # No arg list here since finish might be a script.
 endif
 
 
 ## No test needed for internal helper m-file.
 %!assert (1)
 
diff --git a/scripts/startup/site-rcfile b/scripts/startup/site-rcfile
--- a/scripts/startup/site-rcfile
+++ b/scripts/startup/site-rcfile
@@ -1,7 +1,7 @@
 ## System-wide startup file for Octave.
 ##
 ## If the environment variable OCTAVE_SITE_INITFILE is set when Octave
 ## starts, then that file is executed instead of this file.
 ##
-## This file should contain any commands that should be executed each
-## time Octave starts for every user at this site.
+## This file contain commands that should be executed each time Octave starts
+## for every user at this site.
diff --git a/scripts/startup/version-rcfile b/scripts/startup/version-rcfile
--- a/scripts/startup/version-rcfile
+++ b/scripts/startup/version-rcfile
@@ -1,25 +1,30 @@
 ## System-wide startup file for Octave.
 ##
 ## If the environment variable OCTAVE_VERSION_INITFILE is set when Octave
 ## starts, then that file is executed instead of this file.
 ##
-## This file should contain any commands that should be executed each
-## time Octave starts for every user at this site.
+## This file contains commands that should be executed each time Octave starts
+## for every user at this site.
 
-## Configure readline using the file inputrc in the Octave startup
-## directory.
-
+## Configure readline using the file inputrc in the Octave startup directory.
 readline_read_init_file (sprintf ("%s%s%s",
                                   octave_config_info ("startupfiledir"),
                                   filesep, "inputrc"));
 
+## Configure LESS pager if present
 if (strcmp (PAGER (), "less") && isempty (getenv ("LESS")))
   PAGER_FLAGS ('-e -X -P"-- less ?pB(%pB\\%):--. (f)orward, (b)ack, (q)uit$"');
 endif
 
 ## This appears here instead of in the pkg/PKG_ADD file so that --norc
 ## will also skip automatic loading of packages.
-
 pkg ("load", "auto");
 
+## For Matlab compatibility, run startup.m when starting Octave.
+if (exist ("startup", "file))
+  startup;  # No arg list here since startup might be a script.
+endif
+
+## For Matlab compatibility, schedule finish.m to run when exiting Octave.
 atexit ("__finish__");
+
